src/server/index.js

```diff
@@ -1,188 +1,228 @@
+const sha5 = (data) => crypto.createHash('sha512').update(data).digest()
 
-const checkAckLoop = async (rpc, onEnd) => {
+const checkAckLoop = async (wrappedStream, onEnd) => {
   try {
     while (true) {
-      await rpc.readProto(DiscoveryAck)
-      await sleep(100)
+      await wrappedStream.readPB(DiscoveryAck)
+      await delay(100)
     }
   } catch (err) { // we'll get here after a disconnect
     onEnd()
   }
 }
 
+/**
+* Stardust Transport Server
+* @class
+*/
 class Server {
-  constructor ({ transports, addresses, muxers }) {
-    this.switch = new MicroSwitch({ transports, addresses, muxers, handler: this.handler.bind(this) })
-
+  /**
+   * @constructor
+   * @param {Object} options - Options for the listener
+   * @param {Array<multiaddr>} options.addresses
+   * @param {Array<Transport>} options.transports
+   * @param {Array<Multiplexer>} options.muxers
+   * @param {Array<Encryption>} options.encryption
+   * @param {PeerInfo} options.peerInfo
+   */
+  constructor ({
+    addresses = [multiaddr('/ip6/::/tcp/5892/ws')],
+    transports = [Transport],
+    muxers = [Muxer],
+    encryption = [Secio],
+    peerInfo
+  } = {}) {
+    this.peerAddr = addresses
     this.network = {}
     this.networkArray = []
-    this._emptyCachedDiscovery = this._cachedDiscovery = Buffer.from('01', 'hex')
+    this._cachedDiscovery = {}
 
-    this.switch.protos.addHandler('/p2p/stardust/0.1.0', (protocol, conn) => {
-      log('adding handler for stardust v0.1.0')
-
-      try {
-        this.handlerV010(conn, conn.info.info.muxed)
-      } catch (err) {
-        log(err)
-      }
-    })
+    this.libp2p = undefined
+    this._transports = transports
+    this._muxers = muxers
+    this._encryption = encryption
+    this._peerInfo = peerInfo
   }
 
-  async handler (conn) {
-    log('new connection')
+  removeFromNetwork (client) {
+    log('removing %s from network', client.id.toB58String())
+
+    delete this.network[client.id.toB58String()]
+    this.update()
+    this.broadcastDiscovery()
+  }
 
-    try {
-      const muxed = await this.switch.wrapInMuxer(conn, true) // add muxer ontop of raw socket
-      conn = await newMuxConn(muxed) // get a muxed connection
+  addToNetwork (connection, wrappedStream, id) {
+    log('adding %s to network', id.toB58String())
 
-      conn.muxed = muxed
-      await this.switch.negotiateProtocol(conn)
-    } catch (err) {
-      log(err)
+    const client = this.network[id.toB58String()] = {
+      id,
+      connection,
+      wrappedStream
     }
+
+    checkAckLoop(wrappedStream, () => this.removeFromNetwork(client))
+
+    this.update()
+    this.broadcastDiscovery()
   }
 
-  async handlerV010 (conn, muxed) {
-    const rpc = LP(conn) // turn into length-prefixed rpc interface
+  /**
+   * Update discovery data
+   */
+  update () {
+    log('updating cached data')
+    this.networkArray = Object.keys(this.network).map(b58 => this.network[b58])
```

Comment:
just use `Object.values`
```suggestion
    this.networkArray = Object.values(this.network)
```